content/en/docs/reference/access-authn-authz/extensible-admission-controllers.md

```diff
@@ -796,4 +796,87 @@ webhooks:
+   of a container is not set, set default resource limits.
+
+3. For a `CREATE` pod request, inject a sidecar container with name "foo-sidecar".
+
+In the cases above, the webhook can be safely reinvoked, or admit an object that already has the fields set. The admission
+can be applied multiple times without changing the result beyond the initial application.
+
+#### Example of non-idempotent mutating admission webhooks:
+
+1. For a `CREATE` pod request, inject a sidecar container with name "foo-sidecar"
+   suffixed with the current timestamp (e.g. "foo-sidecar-19700101-000000").
+
+In the case above, reinvoking the webhook can result in the same sidecar being injected multiple times to a pod, each time
+with a different container name. Similarly the webhook can inject duplicated containers when the sidecar already exists in
+the user-provided pod.
+
+### Intercepting all versions of an object
+
+It is recommended that admission webhooks should always intercpet all versions of an object by setting `.webhooks[].matchPolicy`
+to `Equivalent`. It is also recommended that admission webhooks should prefer registering for stable versions of resources.
+Failure to intercept all versions of an object can result in admission policies not being enforced for requests in certain
+versions. See [Matching requests: matchPolicy](#matching-requests-matchpolicy) for examples.
+
+### Availability
+
+It is recommended that admission webhooks should should evaluate as quickly as possible, since they add to API request latency.
+It is encouraged to use a small timeout for webhooks. See [Timeouts](#timeouts) for more detail.
+
+It is recommended that admission webhooks should leverage some format of load-balancing, to provide high availability and
+performance benefits. If a webhook is running within the cluster, you can run multiple webhook backends behind a service
+to leverage the load-balancing that service supports.
+
+### Guaranteeing the final state of the object is seen
+
+Admission webhooks that need to guarantee they see the final state of the object in order to enforce policy
+should use a validating admission webhook, since objects can be modified after being seen by mutating webhooks.
+
+For example, a mutating admission webhook is configured to inject a sidecar container with name "foo-sidecar" on every
+`CREATE` pod requests. A validating admisson webhook must be configurated to intercept `CREATE` pod requests, and validate
+that a container with name "foo-sidecar" exists in the to-be-created object.
+
+### Avoiding deadlocks in self-hosted webhooks
+
+A webhook that enacts policies for webhook mission-critical resources might cause deadlocks for its own deployment.
+
+For example, a mutating admission webhook is configured to admit `CREATE` pod requests only if certain label is set in the
+pod. The webhook server runs in a deployment which doesn't set the label. When a node that runs the webhook server pods
```

Comment:
nit: ... only if a specific label is set in the Pod.
nit: Can you explain a bit more. What label?